src/html/HTMLWriter.jl

```diff
@@ -800,9 +799,34 @@ function render(doc::Documenter.Document, settings::HTML = HTML())
         copy_asset("themes/$(theme).css", doc)
     end
 
-    size_limit_successes = map(collect(keys(doc.blueprint.pages))) do page
+    function page_navnode(page)
         idx = findfirst(nn -> nn.page == page, doc.internal.navlist)
         nn = (idx === nothing) ? Documenter.NavNode(page, nothing, nothing) : doc.internal.navlist[idx]
+        return nn, idx
+    end
+
+    foreach(keys(doc.blueprint.pages)) do page
+        nn, idx = page_navnode(page)
+        @debug "Indexing $(page) [$(repr(idx))]"
+        generate_index!(ctx, nn)
+    end
+
+    # Make an attempt at avoiding spurious changes to the search index hash.
```

Comment:
Would this be accurate?
```suggestion
    # Sort the index to avoid spurious changes to the search index hash caused
    # by difference in the order of entries
```